test_util/test_launch.py

```diff
@@ -0,0 +1,302 @@
+    config_path = tmpdir.join('my_specific_config.yaml')  # test non-default name
+    config_path.write(config_str)
+    config_path = str(config_path)
+    info_path = str(tmpdir.join('my_specific_info.json'))  # test non-default name
+
+    # Now check launcher via CLI
+    check_cli('create --config-path={} --info-path={}'.format(config_path, info_path))
+    # use the info written to disk to ensure JSON parsable
+    info = load_json(info_path)
+    # General assertions about info
+    assert 'type' in info
+    assert 'provider' in info
+    assert 'ssh' in info
+    assert 'user' in info['ssh']
+    assert 'private_key' in info['ssh']
+
+    check_cli('wait --info-path={}'.format(info_path))
+
+    # clear stdout capture
+    capsys.readouterr()
+    check_cli('describe --info-path={}'.format(info_path))
+    # capture stdout from describe and ensure JSON parse-able
+    description = json.loads(capsys.readouterr()[0])
+
+    # general assertions about description
+    assert 'masters' in description
+    assert 'private_agents' in description
+    assert 'public_agents' in description
+
+    check_cli('test --info-path={}'.format(info_path))
+
+    check_cli('delete --info-path={}'.format(info_path))
+
+    return info, description
+
+
+@contextmanager
+def mock_context(*args, **kwargs):
+    yield type('Tunnelled', (object,), {})
+
+
+@pytest.fixture
+def mock_test_runner(monkeypatch):
+    monkeypatch.setattr(ssh.tunnel, 'tunnel', mock_context)
+    monkeypatch.setattr(test_util.runner, 'integration_test', lambda *args, **kwargs: 0)
+
+
+@pytest.fixture
+def mock_ssh_key_path(tmpdir):
+    ssh_key_path = tmpdir.join('ssh_key')
+    ssh_key_path.write(MOCK_SSH_KEY_DATA)
+    return str(ssh_key_path)
+
+
+@pytest.fixture
+def mock_aws_cf_simple_backend(monkeypatch, mock_test_runner):
+    """Does not include SSH key mocking
+    """
+    # mock create
+    monkeypatch.setattr(test_util.aws.BotoWrapper, 'create_stack', lambda *args: None)
+    # mock wait
+    monkeypatch.setattr(test_util.aws.CfStack, 'get_stack_details', lambda _: {'StackStatus': 'CREATE_COMPLETE'})
+    # mock describe
+    monkeypatch.setattr(test_util.aws.DcosCfSimple, 'get_master_ips',
+                        lambda _: [Host('127.0.0.1', '12.34.56')])
+    monkeypatch.setattr(test_util.aws.DcosCfSimple, 'get_private_agent_ips',
+                        lambda _: [Host('127.0.0.1', None)])
+    monkeypatch.setattr(test_util.aws.DcosCfSimple, 'get_public_agent_ips',
+                        lambda _: [Host('127.0.0.1', '12.34.56')])
+    # mock delete
+    monkeypatch.setattr(test_util.aws.DcosCfSimple, 'delete', lambda _: None)
+    monkeypatch.setattr(test_util.aws.BotoWrapper, 'delete_key_pair', lambda *args: None)
+    # mock config
+    monkeypatch.setattr(test_util.aws.BotoWrapper, 'create_key_pair', lambda *args: MOCK_SSH_KEY_DATA)
+
+
+@pytest.fixture
+def mock_aws_cf_simple(mock_aws_cf_simple_backend, mock_ssh_key_path):
```

Comment:
`mock_aws_cf_simple_backend` is not used in this method, do you think using `@pytest.mark.usefixtures` will be preferable? http://doc.pytest.org/en/latest/fixture.html#using-fixtures-from-classes-modules-or-projects

(Basically, that only states states the dependency as a decorator, when we really dont need the object that is sent as argument here)